gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/rolesprovider/github/GithubTeamsUserRolesProvider.groovy

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.security.rolesprovider.github
+
+import com.netflix.spinnaker.gate.config.GitHubProperties
+import com.netflix.spinnaker.gate.security.rolesprovider.UserRolesProvider
+import com.netflix.spinnaker.gate.security.rolesprovider.github.client.GitHubMaster
+
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.InitializingBean
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import org.springframework.util.Assert
+import retrofit.RetrofitError
+import retrofit.client.Response
+
+@Slf4j
+@Component
+@ConditionalOnProperty(value = "auth.groupMembership.service", havingValue = "github")
+class GithubTeamsUserRolesProvider implements UserRolesProvider, InitializingBean {
+
+  @Autowired
+  GitHubMaster master
+
+  @Autowired
+  GitHubProperties gitHubProperties
+
+  @Override
+  Collection<String> loadRoles(String userName) {
+    if (!userName) {
+      return []
+    }
+    // check organization if set.
+    // If organization is unset, all GitHub users can login and have full access
+    // If an organization is set, add it to roles to restrict users to this organization
+    // If organization is set AND requiredGroupMembership set, organization members will have RO access
+    // and requiredGroupMembership members RW access
+    if (!gitHubProperties.organization) {
+      return []
+    }
+
+    Boolean isMemberOfOrg = false
+
+    try {
+      Response response = master.gitHubClient.isMemberOfOrganization(gitHubProperties.organization, userName)
+      isMemberOfOrg = (response.status == 204)
+    } catch (RetrofitError e) {
+      if (e.getKind() == RetrofitError.Kind.NETWORK) {
+        log.error("Could not find the server ${master.baseUrl}", e)
+        return []
+      } else if (e.response.status == 404) {
+        return []
+      }
+    }
+    if (isMemberOfOrg) {
+      List result = []
```

Comment:
move `List result = []` out of the if block, then on L89, move `return result` out of the block and you can delete `return []`